src/catalog/zone_map_catalog.cpp

```diff
@@ -0,0 +1,137 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// zone_map_catalog.cpp
+//
+// Identification: src/
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "catalog/zone_map_catalog.h"
+
+#include "catalog/catalog.h"
+#include "executor/logical_tile.h"
+#include "storage/data_table.h"
+#include "storage/tuple.h"
+#include "type/types.h"
+#include "type/value_factory.h"
+
+namespace peloton {
+namespace catalog {
+
+// Global Singleton : I really dont want to do this and I know this sucks.
+// but #796 is still not merged when I am writing this code and I really
+// am sorry to do this. When PelotonMain() becomes a reality, I will
+// fix this for sure.
+ZoneMapCatalog *ZoneMapCatalog::GetInstance(concurrency::Transaction *txn) {
+  static ZoneMapCatalog zone_map_catalog{txn};
```

Comment:
This is the singleton problem. It wont instantiate the new ZoneMapCatalog because of the static keyword. This is a general problem which will be fixed later about singletons.